### ISSUE TITLE
Handle recursive references nicely

### ISSUE THREAD

**Issue by mikebell** [kevindew/openapi3_parser#4]
Hi Kevin,

The content store api spec (https://raw.githubusercontent.com/alphagov/content-store/deployed-to-production/openapi.yaml) has a circular reference in it (line 221). Would it be possible for the gem to catch these occurrences and cycle back round to fill it in but mark it as done to stop recursion?

@lewisnyman has mentioned that widdershins handles this but I'm not sure on the implementation and whether the logic could be copied.

**Comment by kevindew** [comment]
Oh yes, that is a pain. 

If I remember right there are various points in this that handle recursive references but it looks like something slipped through here which is causing this loop.

I've got an integration test put together for this now and try get together a solution for it - just means refreshing the brain with all the recursion aspects of it.

**Comment by kevindew** [comment]
I've had a bit of a dig into it and it's a bit more problematic than I thought. I've got some stuff together than starts to solve it but may take a few days to find time to sort it out.

**Comment by kevindew** [comment]
Hey @mikebell I've just put out a new release that fixes the recursive issue you were having - as well as a couple of other fixes. So if you update the gem things should be good.

I checked out the content-store schema to test it worked and there's a couple of issues in that file that make it an invalid schema:

1 - https://github.com/alphagov/content-store/blob/master/openapi.yaml#L167 not allowed description here
2 - https://github.com/alphagov/content-store/blob/master/openapi.yaml#L279 - the data should be in a values node

So to test out I created a file with the errors resolved: https://gist.github.com/kevindew/1f1914f237046bd97edc7b0b600bb772

Also this most recent release of the gem does include some pretty substantial refactoring I'd been working on for a while, so if you encounter any new issues do let me know.
